### PR TITLE
YUNIKORN-1557: expose build info in webapp

### PR DESCRIPTION
**Pull request by lixmgl**
### What is this PR for?
A few sentences describing the overall goals of the pull request's commits.
First time? Check out the contributing guide - http://yunikorn.apache.org/community/how_to_contribute   

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### What is the Jira issue?
* Open an issue on Jira https://issues.apache.org/jira/browse/YUNIKORN-1557
* Put link here, and add [YUNIKORN-1557] in PR title, eg. `expose build info in webapp`

### How should this be tested?
tested locally

### Screenshots (if appropriate)
moved buildInfo into a new Info card:
Screenshot 2023-05-09 at 5 51 03 PM
Screenshot 2023-05-09 at 5 51 11 PM

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

**Review by 0yukali0 — CHANGES_REQUESTED** [review]
Almost done in my viewpoint.
BTW, i cloned this PR to run the web locally successfully.
image

File: src/app/models/cluster-info.model.ts [hunk 1]
```diff
@@ -18,8 +18,15 @@
...
     clusterName: string;
     clusterStatus?: string;
 }
+
+export interface BuildInfo {
+    buildDate: string;
+    buildVersion: string;
+    isPluginVersion: string,
+    rmId: string;
+}
```
[the remark this answers is not in the document]
Comment (lixmgl) [reply src/app/models/cluster-info.model.ts]: @0yukali0 thanks for reviewing the pr, I added a new line.

**Review by craigcondit — CHANGES_REQUESTED** [review]
I think it would be best to put that info under the Help (?) menu. It makes the main UI very cluttered, and requires a quite-wide browser window to display properly.

**Comment by lixmgl** [comment]
> 



> I think it would be best to put that info under the Help (?) menu. It makes the main UI very cluttered, and requires a quite-wide browser window to display properly.

@craigcondit I don't think ? menu is the best approach since buildInfo is part of ClusterInfo. I adjust the format to include buildInfo in a new column (please check screenshot #3), this approach addressed quite-wide browser issue. Later we can add more buildInfo into this column.

**Comment by craigcondit** [comment]
> I don't think ? menu is the best approach since buildInfo is part of ClusterInfo. I adjust the format to include buildInfo in a new column (please check screenshot #3), this approach addressed quite-wide browser issue. Later we can add more buildInfo into this column.

It's still information that is only really needed for debugging, not day-to-day use. Most users of the UI are not going to care what build ID or date or whatever are present in the metadata. But an about page that lists information such as this would be helpful. The fact that the REST API returns this as part of ClusterInfo isn't really relevant -- the UI doesn't need to organize the information the same way.

**Comment by lixmgl** [comment]
> 



> > I don't think ? menu is the best approach since buildInfo is part of ClusterInfo. I adjust the format to include buildInfo in a new column (please check screenshot #3), this approach addressed quite-wide browser issue. Later we can add more buildInfo into this column.
> 
> It's still information that is only really needed for debugging, not day-to-day use. Most users of the UI are not going to care what build ID or date or whatever are present in the metadata. But an about page that lists information such as this would be helpful. The fact that the REST API returns this as part of ClusterInfo isn't really relevant -- the UI doesn't need to organize the information the same way.

That make sense @craigcondit 
I created a new Info card for Build related info (Please check screenshot #4 & 5)

> > I don't think ? menu is the best approach since buildInfo is part of ClusterInfo. I adjust the format to include buildInfo in a new column (please check screenshot #3), this approach addressed quite-wide browser issue. Later we can add more buildInfo into this column.
> 
> It's still information that is only really needed for debugging, not day-to-day use. Most users of the UI are not going to care what build ID or date or whatever are present in the metadata. But an about page that lists information such as this would be helpful. The fact that the REST API returns this as part of ClusterInfo isn't really relevant -- the UI doesn't need to organize the information the same way.

Good point @craigcondit, I moved build related info into a new Info card. 
Please check screenshot. If you click on the icon, it will display all build info.

**Comment by wilfred-s** [comment]
I like the use and location of the info card, I think that is also an extensible setup. We are looking at adding more details to the build info and that card can easily expand with further details.

Not sure why the unit tests are failing, not really a js developer...

**Comment by lixmgl** [comment]
> I like the use and location of the info card, I think that is also an extensible setup. We are looking at adding more details to the build info and that card can easily expand with further details.
> 
> Not sure why the unit tests are failing, not really a js developer...

I just fixed the unit tests. 
Right, i will add more buildInfo after backend changes are landed. Tracking ticket: [YUNIKORN-1727](https://issues.apache.org/jira/browse/YUNIKORN-1727)

**Review by craigcondit — COMMENTED** [review]
+1 LGTM.

**Pull request closed without merging**